### PR TITLE
Merge noarxiv_option and compiler_support into develop

### PR DESCRIPTION
**Pull request by johannesjmeyer**
Combined the changes from noarxiv_option and compiler_support deriving from the current develop branch.

### CODE REVIEW

File: quantum-template.tex [hunk 1]
```diff
@@ -167,10 +167,11 @@ \subsection{References and footnotes}
 within the first 5 lines of your main LaTeX file as suggested by the arXiv \cite{arxivpdfoutput}.
 
 \section{Unpublished and accepted works}
-This class supports the \texttt{unpublished} option which disables all Quantum related branding such as the logo, color, and hyperlink in the title.
-It can be used for works not yet accepted in Quantum or not even intended for submission to Quantum.
+This class supports the \texttt{unpublished} option which disables all Quantum related branding such as the logo, color, and hyperlink in the title. It can be used for works not yet accepted in Quantum.
```
Comment (cgogolin) [inline quantum-template.tex]:
```suggestion
This class supports the \texttt{unpublished} option which disables all Quantum related branding such as the logo, color, and hyperlink in the title. It should be used for works not yet accepted in Quantum.
```

File: quantumarticle.cls [hunk 2]
```diff
@@ -219,6 +219,20 @@ class for Quantum - the open journal for quantum science (http://quantum-journal
 \togglefalse{@allowtoday}
 \DeclareOptionX{allowtoday}{\toggletrue{@allowtoday}}
 
+\DeclareOptionX{noarxiv}
+{
+	% Sets a4paper, allowfontchangeintitle, allowtoday, nopdfoutputerror and unpublished
+	\setlength\paperheight{297mm}%
```
Comment (cgogolin) [inline quantumarticle.cls]: Is it possible to use `\ExeuteOptionsX{a4paper}` here instead of manually doing eveything here that `a4paper` would do?

**Review by cgogolin — APPROVED** [review]
Have a look at my very minor changes, otherwise this looks ready to merge.

File: quantumarticle.cls [hunk 3]
```diff
@@ -1353,11 +1371,17 @@ class for Quantum - the open journal for quantum science (http://quantum-journal
 	{%
 	}
 	{	
-		\ifnumequal{\pdfoutput}{0}
-		{%
-			\ClassError{quantumarticle}{Please add \string\pdfoutput=1 within the first 5 lines of the preamble of your main LaTeX file as suggested by the arXiv (https://arxiv.org/help/submit_tex). Without this, the arXiv compiles your document with latex instead of pdflatex and hyperlinks in the bibliography broken over several lines will not work properly. See http://tex.stackexchange.com/questions/182523/publishing-to-arxiv-with-hyperlinks for more information. If you still want to use standard latex (for example because you depend on psfrag) instead of pdflatex despite this problem, you can disable this error by giving the nopdfoutputerror option to quantumarticle. In this case please check carefully that the links in your document, especially in the bibliography work properly.}{}%see also http://tex.stackexchange.com/questions/341466/latex-macros-in-classwarning-or-classerror for why \string is necessary
+		\ifcsdef{pdfoutput}
```
Comment (cgogolin) [inline quantumarticle.cls]: After writing https://github.com/quantum-journal/quantum-journal/issues/75#issuecomment-478358088 I am no longer quite sure whether the code here actually does what it should do: Shouldn't there also be an error if `pdfoutput` is not defined? The current code throws an error if `pdfoutput` is defined and equal to 0. Shouldn't it through an error whenever `pdfoutput` is not defined or not equal to 1? I.e.: 
```
\ifcsdef{pdfoutput}
{
\ifnumequal{\pdfoutput}{1}
{}
{\ClassError{quantumarticle}{Please add...}}
}
{
\ClassError{quantumarticle}{Please add...}
}

```

File: quantumarticle.cls (hunk 2 above)
Comment (johannesjmeyer) [reply quantumarticle.cls]: I was sure that I tried that and it didn't work. But I had a look on the `xkeyval` docs and tried it and it does actually work like a charm. I corrected that.

File: quantumarticle.cls (hunk 3 above)
Comment (johannesjmeyer) [reply quantumarticle.cls]: This is exactly what the code does, plus an additional check for `nopdfoutputerror` (which is basically "i don't care about pdfoutput") around the whole block.

**Comment by cgogolin** [comment]
I just modified the check of pdfoutput to no longer check for =0 but to check for =1 to also catch cases where, for some reason, someone decides to put \pdfoutput=42 in their file...

**Comment by cgogolin** [comment]
Merged. Thanks for the good work!

**Pull request merged**